subiquity/server/controllers/filesystem.py

```diff
@@ -262,6 +263,23 @@ def dd(cls, name: str, min_size: int):
         )
 
 
+def validate_pin_pass(
+    passphrase_allowed: bool, pin_allowed: bool, passphrase: str, pin: str
+) -> None:
+    if passphrase is not None and pin is not None:
+        raise StorageRecoverableError("must supply at most one of pin and passphrase")
+    if not pin_allowed and pin is not None:
+        raise StorageRecoverableError("unexepected pin supplied")
+    if not passphrase_allowed and passphrase is not None:
+        raise StorageRecoverableError("unexepected passphrase supplied")
+
+    if pin is not None:
+        try:
+            int(pin)
```

Comment:
`int()` is quite permissive. For instance `int(+4)` and `int(-3)` are valid. Instead we could use `str.isdecimal`.
```python
if not pin.isdecimal():
    raise
```

Reply:
Much more sensible